http-transport-binding.md

```diff
@@ -0,0 +1,363 @@
+"SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be
+interpreted as described in [RFC2119][RFC2119].
+
+### 1.2. Relation to HTTP
+
+This specification does not prescribe rules constraining the use or handling of
+specific [HTTP methods][RFC7231-Section-4], and it also does not constrain the
+[HTTP target resource][RFC7230-Section-5-1] that is used for transferring or
+soliciting events.
+
+Events can be transferred with all standard or application-defined HTTP request
+methods that support payload body transfers. Events can be also be transferred
+in HTTP responses and with all HTTP status codes that permit payload body
+transfers.
+
+All examples herein that show HTTP methods, HTTP target URIs, and HTTP status
+codes are non-normative illustrations.
+
+This specification also applies equivalently to HTTP/2 ([RFC7540][RFC7540]),
+which is compatible with HTTP 1.1 semantics.
+
+### 1.3. Content Modes
+
+This specification defines two content modes for transferring events:
+*structured* and *binary*. Every compliant implementation SHOULD support both
+modes.
+
+In the *structured* content mode, event metadata attributes and event data are
+placed into the HTTP request or response body using an [event
+format](#14-event-formats).
+
+In the *binary* content mode, the value of the event `data` attribute is placed
+into the HTTP request or response body as-is, with the `contentType` attribute
+value declaring its media type; all other event attributes are mapped to HTTP
+headers.
+
+### 1.4. Event Formats
+
+Event formats, used with the *stuctured* content mode, define how an event is
+expressed in a particular data format. All implementations of this
+specification MUST support the [JSON event format][JSON-format], but MAY
+support any additional, including proprietary, formats.
+
+### 1.5. Security
+
+This specification does not introduce any new security features for HTTP, or
+mandate specific existing features to be used. This specification applies
+identically to [HTTP over TLS]([RFC2818][RFC2818]).
+
+## 2. Use of CloudEvents Attributes
+
+This specification does not further define any of the [CloudEvents][CE] event
+attributes.
+
+Two of the event attributes, `contentType` and `data` are handled specially
+and mapped onto HTTP constructs, all other attributes are transferred as
+metadata without further interpretation.
+
+This mapping is intentionally robust against changes, including the addition
+and removal of event attributes, and also accommodates vendor extensions to the
+event metadata. Any mention of event attributes other than `contentType` and
+`data` is exemplary.
+
+### 2.1. contentType Attribute
+
+The `contentType` attribute is assumed to contain a [RFC2046][RFC2046]
+compliant media-type expression.
+
+### 2.2. data Attribute
+
+The `data` attribute is assumed to contain opaque application data that is
+encoded as declared by the `contentType` attribute.
```

Comment:
This is confusing.  How is the event meta-data encoded then?  this seems to invite multiple kinds of encoding.  `contentType` is pretty ambiguous in the specification -- there was prior discussion that it applied to meta-data attributes as well.  The spec says "Describe the data encoding format" which it seems @clemensv interprets as the encoding of the `data` attribute (and others previously discussed as including all of the event data).  Whatever is decided, please clarify in the spec.

Reply:
I agree we should clarify that in the core spec. The contentType field only makes sense to describe the data field, but for that it is required.

Reply:
@clemensv can you open an issue (or PR if you have text in mind) for this so we don't lose track of it?